.github/scripts/withOnyxMigration.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+
+RED='\033[0;31m'
+ORANGE='\033[0;33m'
+GREEN='\033[0;32m'
+BOLD_WHITE='\033[1;37m'
+
+# Set IFS to newline to properly handle file names with spaces.
+# This makes 'for' loop treat each line as a separate item.
+IFS=$'\n'
+
+# Fetch the main branch
+git fetch origin main --no-tags --depth=1
+
+# Get the list of modified and added files
+modified_files=$(git diff --name-only --diff-filter=AM origin/main HEAD -- '*.ts' '*.tsx')
```

Comment:
actually I'm wondering if we need to check `ts` files at all, I think we can check only tsx